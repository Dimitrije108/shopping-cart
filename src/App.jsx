import { useState, createContext, useContext } from 'react';
import { Outlet, useLocation } from "react-router-dom";
import Header from "./components/Header";
import Homepage from "./components/Homepage";
import Footer from "./components/Footer";

const CartContext = createContext({
  cart: null,
  addToCart: () => {},
});

export function App() {
  const {pathname} = useLocation();
  const [cart, setCart] = useState(0);

  const addToCart = () => {
    setCart(cart + 1);
  }

  return (
    // wrap this up in the cart context
    <CartContext.Provider value={{ cart, addToCart }}>
      <Header />
      {pathname === "/" && <Homepage />}
      <Outlet />
      <Footer />
    </CartContext.Provider>
  )
};

export const useCartContext = () => useContext(CartContext);
