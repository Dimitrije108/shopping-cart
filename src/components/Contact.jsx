import { useState } from 'react';

function Contact() {
  return (
		<div>This is contacts page!</div>
	)
}

export default Contact;