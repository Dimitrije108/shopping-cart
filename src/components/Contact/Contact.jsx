import styles from "./Contact.module.css";

export default function Contact() {
  return (
		<>
			<div className={styles.wrapper}>
				{/* <div  className={styles.backdrop}></div> */}
				<h1>Contact Us</h1>
				<div className={styles.locationCont}>
					<img 
						src="/src/assets/galaxy_map.jpg" 
						alt="Galaxy map" 
						width={500}
						title="Galaxy map"
						className={styles.galaxyMap}
					/>
					<div className={styles.planetAndLogoCont}>
						<img 
							src="/src/assets/corellia_planet.webp" 
							alt="Corellia planet" 
							width={250}
							title="Corellia planet"
							className={styles.planetMap}
						/>
						<img 
							src="/src/assets/Corellian_Engineering_Corporation.svg" 
							alt="Corellia Engineering Corporation logo" 
							width={250}
							title="Corellia Engineering Corporation"
							className={styles.cecLogo}
						/>
					</div>
					<div className={styles.contactCont}>
						<div className={styles.locationCont}>
							<div className={styles.iconCont}>
								<img 
									src="/src/assets/location.svg" 
									alt="location icon" 
									width={30}
									title="Location"
								/>
							</div>
							<div className={styles.location}>
								<p className={styles.address}>Orbital Assembly Facility 7</p>
								<p>Corellia, Corellian System</p>
								<p>Corellia Sector, Core Worlds</p>
							</div>
						</div>
						<div className={styles.emailCont}>
							<div className={styles.iconCont}>
								<img 
									src="/src/assets/email.svg" 
									alt="email icon" 
									width={30}
									title="Email"
								/>
							</div>
							<p>sw_shipyard@corellia.cec</p>
						</div>
					</div>
				</div>
				<div className={styles.contactFormsCont}>
					<div className={styles.largeOrderForm}>

					</div>
					<div className={styles.requestPriceForm}>
						
					</div>
				</div>
				<div className={styles.partnersCont}>

				</div>
			</div>
		</>
	)
};
