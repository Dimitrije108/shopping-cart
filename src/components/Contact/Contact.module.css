.wrapper {
	padding: 3rem 1rem 4rem 1rem;

	display: flex;
	flex-direction: column;
	gap: 1rem;
}

.locationWrapper {
	display: grid;
	grid-template-areas: 
		"heading heading heading"
		"map planet contact";
	grid-template-columns: 2fr 1fr 1fr;
	gap: 1rem;
}

.findUsHeading {
	grid-area: heading;
}

.findUsHeading,
.contactUsHeading,
.partnersHeading {
	font-family: Trench;
	font-size: 2.2rem;
	color: white;
}

.partnersHeading {
	text-align: center;
}

.galaxyMapCont {
	grid-area: map;

	width: 100%;
	height: 100%;
	overflow: hidden;
	aspect-ratio: 16 / 9;
	display: flex;
	justify-content: center;
}

.galaxyMapCont,
.planetMapCont,
.cecLogoCont {
	width: 100%;
	height: 100%;
}

.galaxyMap,
.planetMap,
.cecLogo {
	width: 100%;
	height: 100%;
	padding: 6px;
	border: 1px solid rgb(104, 104, 104);

	object-fit: cover;
	object-position: center;
}

.planetAndLogoCont {
	grid-area: planet;

	display: flex;
	flex-direction: column;
	justify-content: flex-end;
	gap: 1rem;
}

.contactCont {
	grid-area: contact;

	padding: 1rem 0.5rem;
	background-color: rgb(20, 20, 20);
	border: 1px solid rgb(104, 104, 104);

	align-self: flex-end;

	display: grid;
	grid-template-areas: 
		"location"
		"email"
		"webpage";
}

.locationCont {
	grid-area: location;
}

.emailCont {
	grid-area: email;
}

.webpageCont {
	grid-area: webpage;
}

.locationCont,
.emailCont,
.webpageCont {
	display: flex;
}

.emailCont,
.webpageCont {
	margin-top: 1rem;
}

.emailCont,
.webpageCont {
	align-items: end;
}

.location {
	font-size: 1rem;
}

.address {
	font-family: 'DINNext bold';
	font-size: 1.2rem;
}

.emailCont > p {
	display: flex;
	align-items: center;
}

.webpageCont > a {
	color: rgb(68, 102, 255);
	transition: 0.3s color ease;
}

.webpageCont > a:hover {
	color: rgb(37, 77, 255);
}

.iconCont {
	margin-top: 4px;
	margin-right: 12px;
	display: flex;
	flex-direction: column;
}

.iconCont > img{
	width: 30px;
}

.contactFormsCont {
	display: flex;
	gap: 2rem;
}

.largeOrderFormCont,
.requestPriceFormCont {
	width: 100%;
	max-width: 700px;

	padding: 1rem;
	background-color: rgb(20, 20, 20);
	border: 1px solid rgb(104, 104, 104);

	display: flex;
	flex-direction: column;
}

.largeOrderForm,
.requestPriceForm {
	flex: 1;

	display: flex;
	flex-direction: column;
	justify-content: space-between;
}

.contactFormHeading {
	margin: 0.5rem 0 2rem 0;

	font-family: 'DINNext bold';
}

.inputCont {
	margin: 1.3rem 0;

	display: flex;
	flex-direction: column;
}

.inputCont > input {
	margin-top: 4px;
	border-bottom: 1px solid rgb(104, 104, 104);
}

.inputCont > input:focus {
	border-bottom: 1px solid rgb(192, 192, 192);
}

.submitBtnCont {
	flex: 1;

	display: flex;
	align-items: flex-end;
}

.submitFormBtn {
	width: 100%;
	padding: 1rem 0.5rem;

	font-family: 'DINNext bold';

	transition: all 0.5s ease;
}

.messageTextarea,
.selectEl {
	margin-top: 4px;
	background-color: rgb(20, 20, 20);
}

.messageTextarea {
	max-width: 100%;
	min-width: 200px;
	max-height: 150px;
	min-height: 80px;

	padding: 4px 6px;
	border: 1px solid rgb(104, 104, 104);
}

.messageTextarea:focus {
	border: 1px solid rgb(192, 192, 192);
}

.partnersCont {
	display: flex;
	flex-direction: column;
	align-items: center;
}

.logoWrapper {
	border: 1px solid rgb(104, 104, 104);
}

.logoCont {
	margin: 6px;

	background-color: white;

	display: flex;
	justify-content: center;
	align-items: center;
}

.logoCont > a {
	padding: 1rem 2rem;
}

.partnerLogo {
	width: 100px;
}

@media (max-width: 1200px) {
	.locationWrapper {
		grid-template-areas: 
			"heading heading"
			"map planet"
			"contact contact";
		grid-template-columns: 2fr 1fr;
		justify-content: center;
	}

	.contactCont {
		padding: 1rem;

		grid-template-areas: 
		"location email webpage";
		align-items: flex-start;
		gap: 1rem;
	}

	.emailCont,
	.webpageCont {
		margin-top: 0;
	}
}

@media (max-width: 880px) {
	.location {
		font-size: 0.9rem;
	}
	
	.address {
		font-size: 1rem;
	}
	
	.emailCont > p {
		font-size: 0.9rem;
	}

	.webpageCont > .iconCont {
		align-self: flex-start;
	}
	
	.webpageCont > a {
		font-size: 0.9rem;
	}
	
	.iconCont {
		margin-right: 8px;
	}
}

@media (max-width: 768px) {
	.wrapper {
		padding: 2rem 1rem 3rem 1rem;
	}

	.locationWrapper {
		grid-template-areas: 
			"heading"
			"map"
			"planet"
			"contact";
		grid-template-columns: 1fr;
		grid-template-rows: min-content 2fr 1fr 1fr;
		justify-content: center;
	}

	.planetAndLogoCont {
		flex-direction: row;
	}

	.contactCont {
		grid-template-areas: 
		"location email"
		"webpage webpage";
		justify-items: center;
	}

	.contactFormsCont {
		display: flex;
		flex-direction: column;
		gap: 2rem;
	}
}

@media (max-width: 550px) {
	.logoCont {
		flex-direction: column;
	}

	.logoCont > a {
		padding: 1rem;
	}

	.partnerLogo {
		width: 70px;
	}
}

@media (max-width: 500px) {
	.contactCont {
		grid-template-areas: 
			"location"
			"email"
			"webpage";
		justify-items: start;
	}
}
