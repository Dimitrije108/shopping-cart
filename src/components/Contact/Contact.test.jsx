import { describe, it, expect } from "vitest";
import { render, screen  } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Contact from "./Contact";
import { App } from "../../App";

describe("Contact component", () => {
	it("Contact component snapshot", () => {
		const contact = render(
			<MemoryRouter>
				<Contact />
			</MemoryRouter>
		)

		expect(contact).toMatchSnapshot();
	});

	describe("Contact component renders correctly", () => {
		it("Find Us section renders correctly", () => {
			render(
				<MemoryRouter>
					<Contact />
				</MemoryRouter>
			)

			const heading = screen.getByText("Find Us");
			const galaxyMap = screen.getByTitle("Galaxy map");
			const corelliaPlanetImg = screen.getByTitle("Corellia planet");
			const cecLogo = screen.getByTitle("Corellia Engineering Corporation");
			const locationIcon = screen.getByTitle("Location");
			const location = screen.getByText("Orbital Assembly Facility 7");
			const emailIcon = screen.getByTitle("Email");
			const email = screen.getByText("sw_shipyard@corellia.cec");
			const webpageIcon = screen.getByTitle("Webpage");
			const webpageLink = screen.getByTestId("webpageLink");
			const webpage = screen.getByText("Corellian Engineering Corporation");

			expect(heading).toBeInTheDocument();
			expect(galaxyMap).toHaveAttribute("src", "/src/assets/galaxy_map.jpg");
			expect(corelliaPlanetImg).toHaveAttribute("src", "/src/assets/corellia_planet.webp");
			expect(cecLogo).toHaveAttribute("src", "/src/assets/Corellian_Engineering_Corporation.svg");
			expect(locationIcon).toHaveAttribute("src", "/src/assets/location.svg");
			expect(location).toBeInTheDocument();
			expect(emailIcon).toHaveAttribute("src", "/src/assets/email.svg");
			expect(email).toBeInTheDocument();
			expect(webpageIcon).toHaveAttribute("src", "/src/assets/website.svg");
			expect(webpageLink).toHaveAttribute("href", "https://starwars.fandom.com/wiki/Corellian_Engineering_Corporation");
			expect(webpage).toBeInTheDocument();
		});

		it("Contact Us section renders correctly", () => {
			render(
				<MemoryRouter>
					<Contact />
				</MemoryRouter>
			)
			
			const heading = screen.getByText("Contact Us");
			const largeOrderFormHeading = screen.getByText("Large Order Contact Form");
			const firstNameInputs = screen.getAllByPlaceholderText("Anakin");
			const lastNameInputs = screen.getAllByPlaceholderText("Skywalker");
			const emailInputs = screen.getAllByPlaceholderText("AniSkywalker@tatooine.galaxy");
			const companyInputs = screen.getAllByPlaceholderText("Jedi Order");
			const largeOrderRequestInput = screen.getByPlaceholderText("Large order request message...");
			const submitBtns = screen.getAllByRole("button", { name: "Submit" });
			const requestShipPriceHeading = screen.getByText("Request Ship Price");
			const selectInput = screen.getByRole("combobox");

			expect(heading).toBeInTheDocument();
			expect(largeOrderFormHeading).toBeInTheDocument();
			expect(firstNameInputs).toHaveLength(2);
			expect(lastNameInputs).toHaveLength(2);
			expect(emailInputs).toHaveLength(2);
			expect(emailInputs[0]).toHaveAttribute("type", "email");
			expect(companyInputs).toHaveLength(2);
			expect(largeOrderRequestInput).toBeInTheDocument();
			expect(submitBtns).toHaveLength(2);
			expect(requestShipPriceHeading).toBeInTheDocument();
			expect(selectInput).toBeInTheDocument();
		});

		it("Our Partners section renders correctly", () => {
			render(
				<MemoryRouter>
					<Contact />
				</MemoryRouter>
			)
			
			const heading = screen.getByText("Our Partners");
			const cfLink = screen.getByTestId("cfLink");
			const cfLogo = screen.getByTitle("Corellian Federation");
			const kdyLink = screen.getByTestId("kdyLink");
			const kdyLogo = screen.getByTitle("Kuat Drive Yards");
			const fondorLink = screen.getByTestId("fondorLink");
			const fondorLogo = screen.getByTitle("Fondor Shipyards");

			expect(heading).toBeInTheDocument();
			expect(cfLink).toHaveAttribute("href", "https://starwars.fandom.com/wiki/Confederation");
			expect(cfLogo).toHaveAttribute("src", "/src/assets/Corellian_Federation_logo.webp");
			expect(kdyLink).toHaveAttribute("href", "https://starwars.fandom.com/wiki/Kuat_Drive_Yards");
			expect(kdyLogo).toHaveAttribute("src", "/src/assets/Kuat-Drive-Yards.svg");
			expect(fondorLink).toHaveAttribute("href", "https://starwars.fandom.com/wiki/Fondor_Shipyards");
			expect(fondorLogo).toHaveAttribute("src", "/src/assets/Fondor-Shipyards.jpg");
		});
	});
});
