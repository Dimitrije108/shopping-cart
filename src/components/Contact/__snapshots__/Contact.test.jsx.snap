// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Contact component > Contact component snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="
					_layoutWrapper_aadc39 
					_wrapper_3e421e
				"
      >
        <div
          class="_locationWrapper_3e421e"
        >
          <h1
            class="_findUsHeading_3e421e"
          >
            Find Us
          </h1>
          <div
            class="_galaxyMapCont_3e421e"
          >
            <img
              alt="Galaxy map"
              class="_galaxyMap_3e421e"
              src="/src/assets/images/galaxy_map.jpg"
              title="Galaxy map"
            />
          </div>
          <div
            class="_planetAndLogoCont_3e421e"
          >
            <div
              class="_planetMapCont_3e421e"
            >
              <img
                alt="Corellia planet"
                class="_planetMap_3e421e"
                src="/src/assets/images/corellia_planet.webp"
                title="Corellia planet"
              />
            </div>
            <div
              class="_cecLogoCont_3e421e"
            >
              <img
                alt="Corellia Engineering Corporation logo"
                class="_cecLogo_3e421e"
                src="/src/assets/icons/Corellian_Engineering_Corporation.svg"
                title="Corellia Engineering Corporation"
              />
            </div>
          </div>
          <div
            class="_contactCont_3e421e"
          >
            <div
              class="_locationCont_3e421e"
            >
              <div
                class="_iconCont_3e421e"
              >
                <img
                  alt="location icon"
                  src="/src/assets/icons/location.svg"
                  title="Location"
                />
              </div>
              <div
                class="_location_3e421e"
              >
                <p
                  class="_address_3e421e"
                >
                  Orbital Assembly Facility 7
                </p>
                <p>
                  Corellia, Corellian System
                </p>
                <p>
                  Corellia Sector, Core Worlds
                </p>
              </div>
            </div>
            <div
              class="_emailCont_3e421e"
            >
              <div
                class="_iconCont_3e421e"
              >
                <img
                  alt="email icon"
                  src="/src/assets/icons/email.svg"
                  title="Email"
                />
              </div>
              <p>
                sw_shipyard@corellia.cec
              </p>
            </div>
            <div
              class="_webpageCont_3e421e"
            >
              <div
                class="_iconCont_3e421e"
              >
                <img
                  alt="webpage icon"
                  src="/src/assets/icons/website.svg"
                  title="Webpage"
                />
              </div>
              <a
                data-testid="webpageLink"
                href="https://starwars.fandom.com/wiki/Corellian_Engineering_Corporation"
              >
                Corellian Engineering Corporation
              </a>
            </div>
          </div>
        </div>
        <h1
          class="_contactUsHeading_3e421e"
        >
          Contact Us
        </h1>
        <div
          class="_contactFormsCont_3e421e"
        >
          <div
            class="_largeOrderFormCont_3e421e"
          >
            <h2
              class="_contactFormHeading_3e421e"
            >
              Large Order Contact Form
            </h2>
            <form
              action=""
              class="_largeOrderForm_3e421e"
            >
              <div
                class="_inputCont_3e421e"
              >
                <label
                  for="firstName"
                >
                  First name*
                </label>
                <input
                  id="firstName"
                  maxlength="18"
                  minlength="2"
                  name="firstName"
                  placeholder="Anakin"
                  required=""
                />
              </div>
              <div
                class="_inputCont_3e421e"
              >
                <label
                  for="lastName"
                >
                  Last name*
                </label>
                <input
                  id="lastName"
                  maxlength="18"
                  minlength="2"
                  name="lastName"
                  placeholder="Skywalker"
                  required=""
                />
              </div>
              <div
                class="_inputCont_3e421e"
              >
                <label
                  for="mail"
                >
                  Contact email*
                </label>
                <input
                  id="mail"
                  maxlength="40"
                  minlength="2"
                  name="mail"
                  placeholder="AniSkywalker@tatooine.galaxy"
                  required=""
                  type="email"
                />
              </div>
              <div
                class="_inputCont_3e421e"
              >
                <label
                  for="company"
                >
                  Company
                </label>
                <input
                  id="company"
                  maxlength="18"
                  minlength="2"
                  name="company"
                  placeholder="Jedi Order"
                />
              </div>
              <div
                class="_inputCont_3e421e"
              >
                <label
                  for="message"
                >
                  Request*
                </label>
                <textarea
                  class="_messageTextarea_3e421e"
                  id="message"
                  name="message"
                  placeholder="Large order request message..."
                  required=""
                />
              </div>
              <div
                class="_submitBtnCont_3e421e"
              >
                <button
                  class="_submitFormBtn_3e421e"
                  type="submit"
                >
                  Submit
                </button>
              </div>
            </form>
          </div>
          <div
            class="_requestPriceFormCont_3e421e"
          >
            <h2
              class="_contactFormHeading_3e421e"
            >
              Request Ship Price
            </h2>
            <form
              action=""
              class="_requestPriceForm_3e421e"
            >
              <div
                class="_inputCont_3e421e"
              >
                <label
                  for="firstName"
                >
                  First name*
                </label>
                <input
                  id="firstName"
                  maxlength="18"
                  minlength="2"
                  name="firstName"
                  placeholder="Anakin"
                  required=""
                />
              </div>
              <div
                class="_inputCont_3e421e"
              >
                <label
                  for="lastName"
                >
                  Last name*
                </label>
                <input
                  id="lastName"
                  maxlength="18"
                  minlength="2"
                  name="lastName"
                  placeholder="Skywalker"
                  required=""
                />
              </div>
              <div
                class="_inputCont_3e421e"
              >
                <label
                  for="mail"
                >
                  Contact email*
                </label>
                <input
                  id="mail"
                  maxlength="25"
                  minlength="2"
                  name="mail"
                  placeholder="AniSkywalker@tatooine.galaxy"
                  required=""
                  type="email"
                />
              </div>
              <div
                class="_inputCont_3e421e"
              >
                <label
                  for="company"
                >
                  Company
                </label>
                <input
                  id="company"
                  maxlength="18"
                  minlength="2"
                  name="company"
                  placeholder="Jedi Order"
                />
              </div>
              <div
                class="_inputCont_3e421e"
              >
                <label
                  for="ships"
                >
                  Request price*
                </label>
                <select
                  class="_selectEl_3e421e"
                  id="ships"
                  name="ships"
                  required=""
                >
                  <option
                    value=""
                  >
                    Select ship
                  </option>
                  <option
                    value="hTypeNubianYacht"
                  >
                    H-type Nubian yacht
                  </option>
                  <option
                    value="nabooRoyalStarship"
                  >
                    Naboo Royal Starship
                  </option>
                  <option
                    value="republicCruiser"
                  >
                    Republic Cruiser
                  </option>
                  <option
                    value="gr75MediumTransport"
                  >
                    GR-75 Medium Transport
                  </option>
                  <option
                    value="aa9CoruscantFreighter"
                  >
                    AA-9 Coruscant freighter
                  </option>
                  <option
                    value="darthVadersTieFighter"
                  >
                    Darth Vader's TIE fighter
                  </option>
                  <option
                    value="slaveI"
                  >
                    Slave I
                  </option>
                </select>
              </div>
              <div
                class="_submitBtnCont_3e421e"
              >
                <button
                  class="_submitFormBtn_3e421e"
                  type="submit"
                >
                  Submit
                </button>
              </div>
            </form>
          </div>
        </div>
        <h1
          class="_partnersHeading_3e421e"
        >
          Our Partners
        </h1>
        <div
          class="_partnersCont_3e421e"
        >
          <div
            class="_logoWrapper_3e421e"
          >
            <div
              class="_logoCont_3e421e"
            >
              <a
                data-testid="cfLink"
                href="https://starwars.fandom.com/wiki/Confederation"
              >
                <img
                  alt="Corellian Federation icon"
                  class="_partnerLogo_3e421e"
                  src="/src/assets/icons/Corellian_Federation_logo.webp"
                  title="Corellian Federation"
                />
              </a>
              <a
                data-testid="kdyLink"
                href="https://starwars.fandom.com/wiki/Kuat_Drive_Yards"
              >
                <img
                  alt="Kuat Drive Yards icon"
                  class="_partnerLogo_3e421e"
                  src="/src/assets/icons/Kuat-Drive-Yards.svg"
                  title="Kuat Drive Yards"
                />
              </a>
              <a
                data-testid="fondorLink"
                href="https://starwars.fandom.com/wiki/Fondor_Shipyards"
              >
                <img
                  alt="Fondor Shipyards icon"
                  class="_partnerLogo_3e421e"
                  src="/src/assets/icons/Fondor-Shipyards.jpg"
                  title="Fondor Shipyards"
                />
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="
					_layoutWrapper_aadc39 
					_wrapper_3e421e
				"
    >
      <div
        class="_locationWrapper_3e421e"
      >
        <h1
          class="_findUsHeading_3e421e"
        >
          Find Us
        </h1>
        <div
          class="_galaxyMapCont_3e421e"
        >
          <img
            alt="Galaxy map"
            class="_galaxyMap_3e421e"
            src="/src/assets/images/galaxy_map.jpg"
            title="Galaxy map"
          />
        </div>
        <div
          class="_planetAndLogoCont_3e421e"
        >
          <div
            class="_planetMapCont_3e421e"
          >
            <img
              alt="Corellia planet"
              class="_planetMap_3e421e"
              src="/src/assets/images/corellia_planet.webp"
              title="Corellia planet"
            />
          </div>
          <div
            class="_cecLogoCont_3e421e"
          >
            <img
              alt="Corellia Engineering Corporation logo"
              class="_cecLogo_3e421e"
              src="/src/assets/icons/Corellian_Engineering_Corporation.svg"
              title="Corellia Engineering Corporation"
            />
          </div>
        </div>
        <div
          class="_contactCont_3e421e"
        >
          <div
            class="_locationCont_3e421e"
          >
            <div
              class="_iconCont_3e421e"
            >
              <img
                alt="location icon"
                src="/src/assets/icons/location.svg"
                title="Location"
              />
            </div>
            <div
              class="_location_3e421e"
            >
              <p
                class="_address_3e421e"
              >
                Orbital Assembly Facility 7
              </p>
              <p>
                Corellia, Corellian System
              </p>
              <p>
                Corellia Sector, Core Worlds
              </p>
            </div>
          </div>
          <div
            class="_emailCont_3e421e"
          >
            <div
              class="_iconCont_3e421e"
            >
              <img
                alt="email icon"
                src="/src/assets/icons/email.svg"
                title="Email"
              />
            </div>
            <p>
              sw_shipyard@corellia.cec
            </p>
          </div>
          <div
            class="_webpageCont_3e421e"
          >
            <div
              class="_iconCont_3e421e"
            >
              <img
                alt="webpage icon"
                src="/src/assets/icons/website.svg"
                title="Webpage"
              />
            </div>
            <a
              data-testid="webpageLink"
              href="https://starwars.fandom.com/wiki/Corellian_Engineering_Corporation"
            >
              Corellian Engineering Corporation
            </a>
          </div>
        </div>
      </div>
      <h1
        class="_contactUsHeading_3e421e"
      >
        Contact Us
      </h1>
      <div
        class="_contactFormsCont_3e421e"
      >
        <div
          class="_largeOrderFormCont_3e421e"
        >
          <h2
            class="_contactFormHeading_3e421e"
          >
            Large Order Contact Form
          </h2>
          <form
            action=""
            class="_largeOrderForm_3e421e"
          >
            <div
              class="_inputCont_3e421e"
            >
              <label
                for="firstName"
              >
                First name*
              </label>
              <input
                id="firstName"
                maxlength="18"
                minlength="2"
                name="firstName"
                placeholder="Anakin"
                required=""
              />
            </div>
            <div
              class="_inputCont_3e421e"
            >
              <label
                for="lastName"
              >
                Last name*
              </label>
              <input
                id="lastName"
                maxlength="18"
                minlength="2"
                name="lastName"
                placeholder="Skywalker"
                required=""
              />
            </div>
            <div
              class="_inputCont_3e421e"
            >
              <label
                for="mail"
              >
                Contact email*
              </label>
              <input
                id="mail"
                maxlength="40"
                minlength="2"
                name="mail"
                placeholder="AniSkywalker@tatooine.galaxy"
                required=""
                type="email"
              />
            </div>
            <div
              class="_inputCont_3e421e"
            >
              <label
                for="company"
              >
                Company
              </label>
              <input
                id="company"
                maxlength="18"
                minlength="2"
                name="company"
                placeholder="Jedi Order"
              />
            </div>
            <div
              class="_inputCont_3e421e"
            >
              <label
                for="message"
              >
                Request*
              </label>
              <textarea
                class="_messageTextarea_3e421e"
                id="message"
                name="message"
                placeholder="Large order request message..."
                required=""
              />
            </div>
            <div
              class="_submitBtnCont_3e421e"
            >
              <button
                class="_submitFormBtn_3e421e"
                type="submit"
              >
                Submit
              </button>
            </div>
          </form>
        </div>
        <div
          class="_requestPriceFormCont_3e421e"
        >
          <h2
            class="_contactFormHeading_3e421e"
          >
            Request Ship Price
          </h2>
          <form
            action=""
            class="_requestPriceForm_3e421e"
          >
            <div
              class="_inputCont_3e421e"
            >
              <label
                for="firstName"
              >
                First name*
              </label>
              <input
                id="firstName"
                maxlength="18"
                minlength="2"
                name="firstName"
                placeholder="Anakin"
                required=""
              />
            </div>
            <div
              class="_inputCont_3e421e"
            >
              <label
                for="lastName"
              >
                Last name*
              </label>
              <input
                id="lastName"
                maxlength="18"
                minlength="2"
                name="lastName"
                placeholder="Skywalker"
                required=""
              />
            </div>
            <div
              class="_inputCont_3e421e"
            >
              <label
                for="mail"
              >
                Contact email*
              </label>
              <input
                id="mail"
                maxlength="25"
                minlength="2"
                name="mail"
                placeholder="AniSkywalker@tatooine.galaxy"
                required=""
                type="email"
              />
            </div>
            <div
              class="_inputCont_3e421e"
            >
              <label
                for="company"
              >
                Company
              </label>
              <input
                id="company"
                maxlength="18"
                minlength="2"
                name="company"
                placeholder="Jedi Order"
              />
            </div>
            <div
              class="_inputCont_3e421e"
            >
              <label
                for="ships"
              >
                Request price*
              </label>
              <select
                class="_selectEl_3e421e"
                id="ships"
                name="ships"
                required=""
              >
                <option
                  value=""
                >
                  Select ship
                </option>
                <option
                  value="hTypeNubianYacht"
                >
                  H-type Nubian yacht
                </option>
                <option
                  value="nabooRoyalStarship"
                >
                  Naboo Royal Starship
                </option>
                <option
                  value="republicCruiser"
                >
                  Republic Cruiser
                </option>
                <option
                  value="gr75MediumTransport"
                >
                  GR-75 Medium Transport
                </option>
                <option
                  value="aa9CoruscantFreighter"
                >
                  AA-9 Coruscant freighter
                </option>
                <option
                  value="darthVadersTieFighter"
                >
                  Darth Vader's TIE fighter
                </option>
                <option
                  value="slaveI"
                >
                  Slave I
                </option>
              </select>
            </div>
            <div
              class="_submitBtnCont_3e421e"
            >
              <button
                class="_submitFormBtn_3e421e"
                type="submit"
              >
                Submit
              </button>
            </div>
          </form>
        </div>
      </div>
      <h1
        class="_partnersHeading_3e421e"
      >
        Our Partners
      </h1>
      <div
        class="_partnersCont_3e421e"
      >
        <div
          class="_logoWrapper_3e421e"
        >
          <div
            class="_logoCont_3e421e"
          >
            <a
              data-testid="cfLink"
              href="https://starwars.fandom.com/wiki/Confederation"
            >
              <img
                alt="Corellian Federation icon"
                class="_partnerLogo_3e421e"
                src="/src/assets/icons/Corellian_Federation_logo.webp"
                title="Corellian Federation"
              />
            </a>
            <a
              data-testid="kdyLink"
              href="https://starwars.fandom.com/wiki/Kuat_Drive_Yards"
            >
              <img
                alt="Kuat Drive Yards icon"
                class="_partnerLogo_3e421e"
                src="/src/assets/icons/Kuat-Drive-Yards.svg"
                title="Kuat Drive Yards"
              />
            </a>
            <a
              data-testid="fondorLink"
              href="https://starwars.fandom.com/wiki/Fondor_Shipyards"
            >
              <img
                alt="Fondor Shipyards icon"
                class="_partnerLogo_3e421e"
                src="/src/assets/icons/Fondor-Shipyards.jpg"
                title="Fondor Shipyards"
              />
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
